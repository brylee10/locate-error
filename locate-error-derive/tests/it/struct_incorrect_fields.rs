//! Only a source and a location field should be present in a struct.
//! Otherwise, raise a compile error.

use locate_error::Locate;
use locate_error::Location;

#[derive(Locate)]
pub struct ExtraFields {
    #[locate_from]
    // This source field can be named anything
    pub inner: Inner,
    pub location: Location,
    pub extra_field: String,
}

#[derive(Locate)]
pub struct MissingLocation {
    #[locate_from]
    pub inner: Inner,
    pub extra_field: String,
}

#[derive(Locate)]
pub struct MissingSource {
    pub location: Location,
    pub extra_field: String,
}

#[derive(Locate)]
pub struct MultipleSourceFields {
    #[locate_from]
    pub inner: Inner,
    pub location: Location,
    #[locate_from]
    pub extra_field: String,
}

pub struct Inner {
    pub field: String,
}

fn main() {}
