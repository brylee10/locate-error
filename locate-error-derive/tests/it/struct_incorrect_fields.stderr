error: Locate requires structs to have only a 'source' field (with the #[locate_from] attribute) and a 'location' field
 --> tests/it/struct_incorrect_fields.rs:7:10
  |
7 | #[derive(Locate)]
  |          ^^^^^^
  |
  = note: this error originates in the derive macro `Locate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Locate requires structs to have a field named 'location' of type `locate_from::Location`
  --> tests/it/struct_incorrect_fields.rs:16:10
   |
16 | #[derive(Locate)]
   |          ^^^^^^
   |
   = note: this error originates in the derive macro `Locate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Locate requires exactly one field marked with #[locate_from], found 0
  --> tests/it/struct_incorrect_fields.rs:23:10
   |
23 | #[derive(Locate)]
   |          ^^^^^^
   |
   = note: this error originates in the derive macro `Locate` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Locate requires exactly one field marked with #[locate_from], found 2
  --> tests/it/struct_incorrect_fields.rs:29:10
   |
29 | #[derive(Locate)]
   |          ^^^^^^
   |
   = note: this error originates in the derive macro `Locate` (in Nightly builds, run with -Z macro-backtrace for more info)
